### ISSUE TITLE
Doc update regarding predictability of experiments using Seeds

### ISSUE THREAD

**Issue by AakashMallik** [pytorch/pytorch#38541]
## 🚀 Feature
The current official docs don't have specific examples regarding the use of seeds and worker_init_fn in order to attain a certain degree of reproducibility.
Following the suggestion by @ptrblck in the [thread](https://discuss.pytorch.org/t/reproducibility-with-all-the-bells-and-whistles/81097), I would like to add the solutions mentioned in the thread to the official docs.

## Motivation
While participating in a Kaggle competition I ended up spending countless hours of GPU compute on reproducing my results across followup runs of the pipeline but I realized it too late that the number of workers, the seeds **inside** these works all affect the outcome of the run. I searched a lot of discussions and docs to end up realizing how to use the worker_init_fn and what to expect from it. I would not want people to end up where I did so I would like to give them a heads up with whatever I have understood.

## Pitch
As suggested by @ptrblck , I would like to update the [doc](https://pytorch.org/docs/stable/notes/randomness.html) with the code examples and give people a clear heads-up regarding the dependency of the results on the workers and seeds and how they may be able to reach an acceptable level of reproducibility.

## Alternatives
As far as I can tell, this would be the most thorough example.

## Additional context
N/A

**Comment by SsnL** [comment]
Yeah it would be good to cross reference among 
+ the reproducibility note https://pytorch.org/docs/stable/notes/randomness.html
+ faq section on data loading randomness https://pytorch.org/docs/stable/notes/faq.html#my-+ data-loader-workers-return-identical-random-numbers
data loading doc on randomness https://pytorch.org/docs/stable/data.html#randomness-in-multi-process-data-loading

**Comment by AakashMallik** [comment]
> Yeah it would be good to cross reference among
> 
> * the reproducibility note https://pytorch.org/docs/stable/notes/randomness.html
> * faq section on data loading randomness https://pytorch.org/docs/stable/notes/faq.html#my-data-loader-workers-return-identical-random-numbers
> * data loading doc on randomness https://pytorch.org/docs/stable/data.html#randomness-in-multi-process-data-loading

Yup, I remember reading each one of these links and finally coming up with a complete code example. My intention was to add this to the docs as a small example for developers to use right out of the docs. 

I saw lots of developers/researchers asking this very question in discussions from time to time with no concrete or wrong examples to go with. I remember @ptrblck being part of a few of these discussions.

**Comment by AakashMallik** [comment]
@SsnL 

> > Yeah it would be good to cross reference among
> > 
> > * the reproducibility note https://pytorch.org/docs/stable/notes/randomness.html
> > * faq section on data loading randomness https://pytorch.org/docs/stable/notes/faq.html#my-data-loader-workers-return-identical-random-numbers
> > * data loading doc on randomness https://pytorch.org/docs/stable/data.html#randomness-in-multi-process-data-loading
> 
> Yup, I remember reading each one of these links and finally coming up with a complete code example. My intention was to add this to the docs as a small example for developers to use right out of the docs.
> 
> I saw lots of developers/researchers asking this very question in discussions from time to time with no concrete or wrong examples to go with. I remember @ptrblck being part of a few of these discussions.

**Comment by mruberry** [comment]
This might be appropriate as another section in our Reproducibility note (https://pytorch.org/docs/master/notes/randomness.html).
